dataflow/main.py

```diff
@@ -20,88 +21,151 @@
     SegmentFeatureTransformDoFn
 
 
-def get_tabel_spec(table_name: str, partition: int = None):
-    return bigquery.TableReference(
-        projectId='cloud-city-cal',
-        datasetId='cloud_city',
-        tableId=f"{table_name}${partition}" if partition is not None else f"{table_name}")
+def get_table_query(table_name: str, start: datetime, end: datetime, buffer: timedelta):
+    """Fetch data with extra buffer added to count for windowing, so that at start time, enough data is available."""
+    return (f"SELECT * FROM `{os.environ['PROJECT_ID']}.{os.environ['DATASET_ID']}.{table_name}` "
+            f"WHERE publish_time >= {int((start - buffer).timestamp() * 1e6)} AND publish_time < {int(end.timestamp() * 1e6)}")
 
 
-def run(partition: int, metadata_version: int, save_to_bigquery: bool, pipeline_args=None):
-    print(f"Processing data for partition {partition} using metadata version {metadata_version}. "
-          f"Output to BigQuery enabled: {save_to_bigquery}.")
+def get_tabel_spec(table_name: str):
+    return bigquery.TableReference(
+        projectId=os.environ['PROJECT_ID'],
+        datasetId=os.environ['DATASET_ID'],
+        tableId=table_name)
+
+
+def run(start: datetime, end: datetime, metadata_version: int,
```

Comment:
Not sure why the table names have to be passed as argument, but either way it looks like a good interface.

Reply:
See comments below. It was designed to accomodate multiple simulations.